lib/ansible/modules/cloud/vmware/vmware_guest_facts.py

```diff
@@ -70,7 +70,7 @@
    datacenter:
      description:
      - Destination datacenter for the deploy operation
-     required: True
+     required: False
```

Comment:
Datacenter parameter is used in [`get_vm`](https://github.com/ansible/ansible/blob/a62f0a2fdafc283203ff62a2a2de3c83916cdee1/lib/ansible/module_utils/vmware.py#L917) API, so I would say this is a required parameter.

Reply:
It works without, but it may be an unexpected side effect.

Could we default the `datacenter` value to `ha-datacenter`? This way we don't enforce the presence of the parameter and we come with reasonable default value instead.

Reply:
Setting default value to `ha-datacenter` results in unwanted outputs as people tend to forget to provide value  as it is not required.

Reply:
I see, that's a good point.